### PR TITLE
zone.ee DNS api support and documentation added

### PR DESCRIPTION
**Pull request by simpss**
Added support for zone.ee DNS API and documentation where I noticed references.  

If there are any more, let me know and I'll update the docs elsewhere as well.

### CODE REVIEW

**Comment by simpss** [comment]
Looks like i'm going to need a little pointer here to pass the travis check. Not very proficient with sh actually :)

I can't seem to figure out why this step is failing:
```
The command "if [ "$TRAVIS_OS_NAME" = "linux" ]; then git diff --exit-code && echo "shfmt OK" ; fi" exited with 1.
```
It just seems to output part of a diff and if i run `git diff --exit-code && echo "shfmt OK"` locally, it outputs "shfmt OK". The only reason for the failure I could figure was that travis isn't using linux for this build? Seems weird.

If anyone has any pointers, let me know and I'll gladly fix it.

File: dnsapi/dns_zoneee.sh [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    return 1
+  fi
+
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _debug "Getting txt records"
+  _zoneee_rest GET "$_domain/txt"
+
+  if printf "%s" "$response" | grep \"result\":error >/dev/null; then
```
Comment (Neilpang) [inline dnsapi/dns_zoneee.sh]: please use `echo` instead of `printf %s`  here.

Comment (simpss) [reply dnsapi/dns_zoneee.sh]: Done, and improved the error message a bit as well.

File: dnsapi/dns_zoneee.sh [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  _zoneee_rest GET "$_domain/txt"
+
+  if echo "$response" | grep \"result\":\"error\" >/dev/null; then
+    _err "Error Getting TXT records"
+    return 1
+  fi
+
+  _info "Adding record"
+  if _zoneee_rest POST "$_domain/txt" "{\"name\":\"$fulldomain\",\"destination\":\"$txtvalue\"}"; then
+    if printf -- "%s" "$response" | grep "$fulldomain" >/dev/null; then
```
Comment (Neilpang) [inline dnsapi/dns_zoneee.sh]: use `echo`

File: dnsapi/dns_zoneee.sh [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    return 1
+  fi
+
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  _debug "Getting txt records"
+  _zoneee_rest GET "$_domain/txt"
+
+  if printf "%s" "$response" | grep \"result\":error >/dev/null; then
```
Comment (Neilpang) [inline dnsapi/dns_zoneee.sh]: echo

File: dnsapi/dns_zoneee.sh [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    _err "Error"
+    return 1
+  fi
+
+  if ! printf "%s" "$response" | grep "\"name\":\"$fulldomain\"" >/dev/null; then
```
Comment (Neilpang) [inline dnsapi/dns_zoneee.sh]: echo

File: dnsapi/dns_zoneee.sh [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+  fi
+}
+
+####################  Private functions below ##################################
+_get_root() {
+  domain=$1
+  i=2
+  p=1
+  while true; do
+    h=$(printf "%s" "$domain" | cut -d . -f $i-100)
```
Comment (Neilpang) [inline dnsapi/dns_zoneee.sh]: echo

File: dnsapi/dns_zoneee.sh [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+      #not valid
+      return 1
+    fi
+
+    if ! _zoneee_rest GET "$h"; then
+      return 1
+    fi
+
+    if _contains "$response" "\"identificator\":\"$h\"" >/dev/null; then
+      _sub_domain=$(printf "%s" "$domain" | cut -d . -f 1-$p)
```
Comment (Neilpang) [inline dnsapi/dns_zoneee.sh]: echo

**Comment by Neilpang** [comment]
please fix the conflicts.

**Pull request closed without merging**